pkg/satoken/token_manager_test.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is a modified version of
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/token/token_manager_test.go
+
+package satoken
```

Comment:
not sure if we can move this into satoken_test

Reply:
Lemme give it a shot

Reply:
We mock out `getToken` which is a private function of `Manager`. Mocking out the call to k8s is more complicated if we go the `_test` package route.

We'd also have to make the `clock` public which should not be public.